File: 1684. Count the Number of Consistent Strings/countConsistentStrings.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
const countConsistentStrings = (allowed, words) => {
  let count = 0;
  const allowedChars = new Set(allowed);

  for (let i = 0; i < words.length; i++) {
    let isConsistent = true;
    for (let j = 0; j < words[i].length; j++) {
      if (!allowedChars.has(words[i][j])) {
        isConsistent = false;
        break;
      }
    }
    if (isConsistent) {
      count++;
    }
  }

  return count;
};


const allowed = "ab";
const words = ["ad","bd","aaab","baa","badab"];
console.log("countConsistentStrings: ", countConsistentStrings(allowed, words)); // 2

File: 1684. Count the Number of Consistent Strings/countConsistentStrings.py
from typing import List

class Solution:
  def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
    count = 0
    allowedChars = set(allowed)
    
    for word in words:
      isConsistent = True
      for c in word:
        if c not in allowedChars:
          isConsistent = False
          break
      if isConsistent:
        count += 1
    
    return count
  

allowed = "ab"
words = ["ad", "bd", "aaab", "baa", "badab"]
solution = Solution()
print("countConsistentStrings: ", solution.countConsistentStrings(allowed, words)) # 2
